libraries/AP_HAL_Linux/AnalogIn_Imx_IOMCU.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "AP_HAL_Linux.h"
+#include "Imx_IOMCU.hpp"
+
+extern const AP_HAL::HAL &hal;
+
+class AnalogSource_Imx_IOMCU : public AP_HAL::AnalogSource {
+public:
+	friend class AnalogIn_Imx_IOMCU;
+	AnalogSource_Imx_IOMCU(int16_t pin)
+	{
+		_pin = pin;
+		_value = 0;
+	}
+	float read_average()
+	{
+		return read_latest();
+	}
+	float read_latest()
+	{
+		return _value;
+	}
+	void set_pin(uint8_t p)
+	{
+		_pin = p;
+	}
+	void set_stop_pin(uint8_t p) {}
+	void set_settle_time(uint16_t settle_time_ms) {}
+	float voltage_average()
+	{
+		return _value;
+	}
+	float voltage_latest()
+	{
+		return _value;
+	}
+	float voltage_average_ratiometric()
+	{
+		return _value;
+	}
+private:
+	int16_t _pin;
+	float _value;
+};
```

Comment:
I'm confused here... `_value` is never updated

Reply:
_value is updated by timer callback function -> "void _update();"